addon/components/amount-input.js

```diff
@@ -15,38 +15,41 @@ const KEY_CODE_COMMA = 188
   @class AmountInput
   @public
 */
-export default Component.extend({
-  layout,
-  classNames: ['amount-input'],
-
+export default class AmountInput extends Component {
   /**
     The currency displayed in the input
     @argument currency
     @type String?
   */
-  currency: 'EUR',
+  get currency() {
+    return this.args.currency ?? 'EUR'
+  }
 
   /**
     A custom class applied on the input
     @argument inputClass
     @type String?
   */
-  inputClass: null,
```

Comment:
Keep comment for auto doc generation, run locally and see at http://localhost:4200/docs/api/components/amount-input